migrations/1_initial_migration.js

```diff
@@ -10,6 +32,58 @@ module.exports = async function(deployer, _, [account]) {
 	}
-	await deployer.deploy(Migrations)
-	await deployer.deploy(DaiHrd)
+}
+
+async function deployUniswapFactory(deployer) {
+	const uniswapExchangeTemplate = await deployer.deploy(artifacts.require('UniswapExchangeTemplate'))
+	const uniswapFactory = await deployer.deploy(artifacts.require('UniswapFactory'))
+	await uniswapFactory.initializeFactory(uniswapExchangeTemplate.address)
+	return uniswapFactory
+}
+
+async function deployDaiHrd(deployer, dai, vat, pot, daiJoin, uniswapFactory) {
+	const uniswapExchangeAddress = await uniswapFactory.createExchange.call('0xadd12e55add12e55add12e55add12e55add12e55')
+
+	await fs.writeFile(path.join(__dirname, '..', 'contracts', 'RuntimeConstants.sol'),
+`// THIS FILE IS AUTOGENERATED DURING MIGRATION, DO NOT EDIT BY HAND
+pragma solidity 0.5.12;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+interface IVat {
+	function dai(address holder) external returns(uint256 attorontodai);
+	function move(address holder, address recipient, uint256 attorontodai) external;
+}
+
+interface IDaiJoin {
+	function live() external returns (uint256 boolean);
+	function join(address holder, uint256 attodai) external;
+	function exit(address holder, uint256 attodai) external;
+}
+
+interface IPot {
+	function pie(address holder) external returns(uint256);
+	function chi() external returns(uint256);
+	function rho() external returns (uint256);
+	function drip() external;
+	function join(uint256 amount) external;
+	function exit(uint256 amount) external;
+}
+
+contract RuntimeConstants {
```

Comment:
Probably worth isolating the dynamic contract here from the static interfaces above, and including them from a separate contract? Just to keep as much as we can outside of JS strings, but not a blocker by any means

Reply:
Good idea, will change.